Core/NoOpRunner.Core/Shapes/PowerUp.cs

```diff
@@ -0,0 +1,31 @@
+﻿using NoOpRunner.Core.Entities;
+using NoOpRunner.Core.Enums;
+using System;
+
+namespace NoOpRunner.Core.Shapes
+{
+    public class PowerUp : BaseShape
+    {
+        public PowerUp(int centerPosX, int centerPosY) : base(centerPosX, centerPosY)
+        {
+            Color powerup = GetRandomPowerUp();
+
+            MapShapeX(0, 2, 1, powerup);
+        }
+
+        private Color GetRandomPowerUp()
```

Comment:
This class constructor doesn't make much sense. Right now you can't create a specific powerup on demand, because the constructor just rolls a random one. The powerup type should be passed as a constructor parameter and should be accessible as a readonly variable, because later on we will have to use that variable to base the pick up logic on it.